cli/cmd/report_rules.go

```diff
@@ -0,0 +1,399 @@
+	reportRulesCommand.AddCommand(reportRulesCreateCommand)
+	reportRulesCommand.AddCommand(reportRulesDeleteCommand)
+}
+
+func buildReportRuleDetailsTable(rule api.ReportRule) string {
+	var (
+		details       [][]string
+		notifications [][]string
+		updatedTime   string
+	)
+	severities := api.NewReportRuleSeveritiesFromIntSlice(rule.Filter.Severity).ToStringSlice()
+
+	if nano, err := strconv.ParseInt(rule.Filter.CreatedOrUpdatedTime, 10, 64); err == nil {
+		updatedTime = time.Unix(nano/1000, 0).Format(time.RFC3339)
+	}
+	details = append(details, []string{"SEVERITIES", strings.Join(severities, ", ")})
+	details = append(details, []string{"DESCRIPTION", rule.Filter.Description})
+	details = append(details, []string{"UPDATED BY", rule.Filter.CreatedOrUpdatedBy})
+	details = append(details, []string{"LAST UPDATED", updatedTime})
+
+	detailsTable := &strings.Builder{}
+	detailsTable.WriteString(renderOneLineCustomTable("ALERT RULE DETAILS",
+		renderCustomTable([]string{}, details,
+			tableFunc(func(t *tablewriter.Table) {
+				t.SetBorder(false)
+				t.SetColumnSeparator(" ")
+				t.SetAutoWrapText(false)
+				t.SetAlignment(tablewriter.ALIGN_LEFT)
+			}),
+		),
+		tableFunc(func(t *tablewriter.Table) {
+			t.SetBorder(false)
+			t.SetAutoWrapText(false)
+		}),
+	),
+	)
+
+	notifcationsMap := rule.ReportNotificationTypes.ToMap()
+	// sort keys
+	keys := make([]string, 0, len(notifcationsMap))
+	for k := range notifcationsMap {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	for _, key := range keys {
+		enabled := strings.Title(strconv.FormatBool(notifcationsMap[key]))
+		notifications = append(notifications, []string{key, enabled})
+	}
+
+	detailsTable.WriteString(renderCustomTable([]string{"NOTIFICATION TYPES", "ENABLED"}, notifications,
+		tableFunc(func(t *tablewriter.Table) {
+			t.SetBorder(false)
+			t.SetColumnSeparator(" ")
+			t.SetAutoWrapText(false)
+		}),
+	),
+	)
+	detailsTable.WriteString("\n")
+
+	if len(rule.EmailAlertChannels) > 0 {
+		channels := [][]string{{strings.Join(rule.EmailAlertChannels, "\n")}}
+		detailsTable.WriteString(renderCustomTable([]string{"EMAIL ALERT CHANNELS"}, channels,
+			tableFunc(func(t *tablewriter.Table) {
+				t.SetBorder(false)
+				t.SetColumnSeparator(" ")
+			}),
+		),
+		)
+		detailsTable.WriteString("\n")
+	}
+
+	if len(rule.Filter.ResourceGroups) > 0 {
+		resourceGroups := [][]string{{strings.Join(rule.Filter.ResourceGroups, "\n")}}
+		detailsTable.WriteString(renderCustomTable([]string{"RESOURCE GROUPS"}, resourceGroups,
+			tableFunc(func(t *tablewriter.Table) {
+				t.SetBorder(false)
+				t.SetColumnSeparator(" ")
+			}),
+		),
+		)
+	}
+
+	return detailsTable.String()
+}
+
+func promptCreateReportRule() (api.ReportRuleResponse, error) {
+	channelList, channelMap := getEmailAlertChannels()
+	notificationFields := structs.Names(api.ReportRuleNotificationTypes{})
+	notificationsMap := make(map[string]bool)
+
+	if len(channelList) < 1 {
```

Comment:
I am worried that `getEmailAlertChannels()` can return `nil, nil` in which case this `len(nil)` will panic.

Reply:
The len of a nil slice is 0

Reply:
Tested this to confirm, when nil we enter the if condition block and return err

Reply:
ok